### PR TITLE
fix random failures on cancel test

### PR DESCRIPTION
**Pull request by samlesu**
This may not be the best solution to test the different behaviour when cancelling a compute plan.

But this will stop the random failures as a quick term solution.

### CODE REVIEW

**Comment by jmorel** [comment]
With this fix, the test really only checks that the execution doesn't fail, which makes it a bit pointless.

I understand the frustration of the test not passing due to timing issue, but if we don't want to rely on timing tuning to limit the occurrences of the issue, then maybe we should just remove the test altogether? 

I'd rather we don't and instead add a `sleep` to the execution of the algos used for this test, but it'll make the test suite a bit slower and I'd understand if this is a big no no.

**Comment by samlesu** [comment]
I think we all agree that:
- we cannot have a test failing randomly
- we want an end-to-end test checking the cancel feature

I do not agree with the fact that the test is now pointless: it is showing that a compute plan status can change from 'canceled' to 'done'.

In my opinion, a big no no is having a test failing randomly. In order to keep going, I will add more triantuples. Would that be ok?

File: tests/test_execution_compute_plan.py [hunk 1]
```diff
@@ -331,7 +342,8 @@ def test_execution_compute_plan_canceled(global_execution_env):
     assert cp.status == assets.Status.canceled
 
     cp = cp.future().wait()
-    assert cp.status == assets.Status.canceled
+    assert cp.status in assets.Status.canceled
```
Comment (jmorel) [inline tests/test_execution_compute_plan.py]: This will work by why the change here?

Comment (samlesu) [reply tests/test_execution_compute_plan.py]: Good catch. Looks like an invalid syntax.
